fgrpc/grpcrunner.go

```diff
@@ -299,3 +313,23 @@ func grpcDestination(dest string) (parsedDest string) {
 	parsedDest += fnet.NormalizePort(port)
 	return parsedDest
 }
+
+// extractDialOptions extract special MD and convert them to dial options.
+func extractDialOptions(in metadata.MD) (out []grpc.DialOption) {
+	for k, v := range in {
+		switch k {
+		case "user-agent":
+			delete(in, k)
+			if v[0] == jrpc.UserAgent {
```

Comment:
any idea why both user agents are present when set?

```
$ go run . grpcping -loglevel debug -c 1 -n 1 -payload foo -grpc -ping -H fOo:bAr -H host:foo-bar -H user-agent:test123 127.0.0.1:8079
```
yields (`go run . server -loglevel debug&` with my addition to ping srv logging of meta data:
```
15:37:31 V pingsrv.go:48> Ping called {Seq:1 Ts:1673653051146622000 Payload:foo DelayNanos:0} (meta map[:authority:[foo-bar] content-type:[application/grpc] foo:[bAr] user-agent:[test123 grpc-go/1.52.0]])
```

Reply:
if you have the answer please put it in #680